providers/implementations/ciphers/cipher_enull_hmac_hw.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019-2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* enull_hmac cipher implementation */
+
+/*
+ * HMAC low level APIs are deprecated for public use, but still ok for internal
+ * use.
+ */
+#include "internal/deprecated.h"
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include "cipher_enull_hmac.h"
+
+static int enull_hmac_initkey(PROV_CIPHER_CTX *bctx, const uint8_t *key,
+                              size_t keylen)
+{
+    PROV_ENULL_HMAC_CTX *ctx = (PROV_ENULL_HMAC_CTX *)bctx;
+
+    if (key == NULL || keylen > sizeof(ctx->key))
+        return 0;
+
+    memcpy(ctx->key, key, keylen);
+    ctx->keylen = keylen;
+    return 1;
+}
+
+static int enull_hmac_initiv(PROV_CIPHER_CTX *bctx,
+                             const unsigned char *iv, size_t ivlen)
+{
+    PROV_ENULL_HMAC_CTX *ctx = (PROV_ENULL_HMAC_CTX *)bctx;
+
+    if (iv == NULL || ivlen > ENULL_HMAC_MAX_IVLEN)
+        return 0;
```

Comment:
thrown an error such as `ERR_R_PASSED_INVALID_ARGUMENT` in this case?

Reply:
throwing `ERR_R_INTERNAL_ERROR`, as it seems more appropriate.